libsrc/telestrat/cgetc.s

```diff
@@ -5,24 +5,38 @@
 
     .import    cursor
 
+    .export    store_char
+
     .include   "telestrat.inc"
 
 .proc _cgetc
-        ; this routine could be quicker if we wrote in page 2 variables,
-        ; but it's better to use telemon routine in that case, because telemon can manage 4 I/O
-        ldx     cursor       ; if cursor equal to 0, then switch off cursor
+        ; This routine could be quicker if we wrote in page 2 variables,
+        ; But it's better to use telemon routine in that case, because telemon can manage 4 I/O
+        ldx     cursor       ; If cursor equal to 0, then switch off cursor
         beq     switchoff_cursor
 
         ldx     #$00         ; x is the first screen
-        BRK_TELEMON(XCSSCR)  ; display cursor
-        jmp     loop         ; could be replaced by a bne/beq but 'jmp' is cleaner than a bne/beq which could expect some matters
+        BRK_TELEMON(XCSSCR)  ; Display cursor
+        jmp     start        ; Could be replaced by a bne/beq but 'jmp' is cleaner than a bne/beq which could expect some matters
 
 switchoff_cursor:
-        ; at this step X is equal to $00, X must be set, because it's the id of the screen (telestrat can handle 4 virtuals screen)
-        BRK_TELEMON(XCOSCR)  ; switch off cursor
+        ; At this step X is equal to $00, X must be set, because it's the id of the screen (telestrat can handle 4 virtuals screen)
+        BRK_TELEMON(XCOSCR)  ; Switch off cursor
+
 
-loop:
-        BRK_TELEMON XRD0     ; waits until key is pressed
-        bcs     loop
+start:
+        lda     store_char   ; Does kbhit store a value in store_char ?
+        bne     @out         ; Yes, we returns A and we reset store_char
+@wait_key:
+        BRK_TELEMON XRD0     ; Waits until key is pressed
+        bcs     @wait_key
+        ldx     #$00
+        rts
+@out:
+        ldx     #$00
+        stx     store_char
         rts
 .endproc
+.data
+store_char:
+        .byte 0
```

Comment:
You can save a byte by putting this into .bss (which is always initialized to 0 by crt0)